PowerPointLabs/PowerPointLabs/PasteLab/PasteLabMain.cs

```diff
@@ -237,5 +260,18 @@ public static void PasteToPosition(Models.PowerPointSlide slide, bool clipboardI
                 }
             }
         }
+
+        private static ShapeRange RemovePlaceholders(PowerPointSlide slide, ShapeRange shapes)
```

Comment:
Ok, i added
- GetShapeWhenTypeMatches 
- GetShapeWhenTypeNotMatches 

Tried not to use the word Select since it does not call Shape.Select()